script/helperFunctions.sh

```
@@ -320,256 +195,143 @@ function findContractInMasterLog() {
-    # Initialize the highest version variable
-    HIGHEST_VERSION="0.0.0"
-
-    # Iterate over each version in the array
-    for VERSION in "${VERSIONS[@]}"; do
-      # Compare the current version with the highest version found so far
-      if [[ "$VERSION" > "$HIGHEST_VERSION" ]]; then
-        HIGHEST_VERSION="$VERSION"
-      fi
-    done
+  echoDebug "Querying MongoDB for getHighestDeployedContractVersionFromMasterLog: $CONTRACT"
+  local MONGO_RESULT=$(bun script/deploy/query-deployment-logs.ts filter \
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)